packages/ethereum-contracts/contracts/agreements/ConstantFlowAgreementV1.sol

```diff
@@ -470,12 +470,21 @@ contract ConstantFlowAgreementV1 is
 
         _requireAvailableBalance(flowVars.token, flowVars.sender, currentContext);
 
-        // for now, we are aware that this will break super tokens with custom super token logic
-        // which choose not to upgrade, in order to fix this, we must use the old try/catch logic
-        // which was previously in place for the previous marketing NFT
-        ISuperToken(
-            address(flowVars.token)
-        ).constantOutflowNFT().onCreate(flowVars.sender, flowVars.receiver);
+        uint256 gasLeftBefore = gasleft();
+    
+        try
+            ISuperToken(address(flowVars.token)).constantOutflowNFT().onCreate{
+                gas: CFA_HOOK_GAS_LIMIT
+            }(flowVars.sender, flowVars.receiver)
+        // solhint-disable-next-line no-empty-blocks
+        {} catch {
+// If the CFA hook actually runs out of gas, not just hitting the safety gas limit, we revert the whole transaction.
+// This solves an issue where the gas estimaton didn't provide enough gas by default for the CFA hook to succeed.
+// See https://medium.com/@wighawag/ethereum-the-concept-of-gas-and-its-dangers-28d0eb809bb2
+            if (gasleft() <= gasLeftBefore / 63) {
```

Comment:
We should finally create a library function for this, e.g.

```
function _revertWhenOutOfGas() internal {
           if (gasleft() <= gasLeftBefore / 63) {
                revert CFA_HOOK_OUT_OF_GAS();
            }
}
```

Reply:
addressed, added a `SafeGasLibrary` lib, open to any new name suggestions